libshaderc_util/src/compiler_test.cc

```diff
@@ -187,4 +187,27 @@ TEST_F(CompilerTest, DISABLED_RespectTargetEnvOnVulkanShader) {
   // TODO(dneto): Add test for a shader that should only compile for Vulkan.
 }
 
+TEST_F(CompilerTest, AddMacroDefinition) {
+  const std::string kMinimalExpandedShader = "void E(){}";
+  compiler_.AddMacroDefinition("E", 1u, "main", 4u);
+  EXPECT_TRUE(SimpleCompilationSucceeds(kMinimalExpandedShader, EShLangVertex));
+}
+
+TEST_F(CompilerTest, AddValuelessMacroDefinition) {
+  const std::string kMinimalExpandedShader =
+      "#ifdef E\n"
+      "void main(){}\n"
+      "#else\n"
+      "#error\n"
+      "#endif";
+  compiler_.AddMacroDefinition("E", 1u, nullptr, 0u);
```

Comment:
I wouldn't set them both to zero.  We should be able to handle either of them being zero.